src/api/api.js

```diff
@@ -115,9 +115,31 @@ function client_factory_from_schema(schema) {
     };
 }
 
+function get_protocol_port(protocol) {
+    switch (protocol.toLowerCase()) {
+        case 'http:':
+        case 'ws:': {
+            return '80';
+        }
+        case 'https:':
+        case 'wss:': {
+            return '443';
+        }
+        default: {
+            return '';
+        }
+    }
+}
```

Comment:
I think a more clear solution is to return the actual port number and not a string representation and `undefined` for the default case. (just `reutrn;`)
so line 142 could be written as:
```
const ports = get_default_ports(Number(port || get_protocol_port(protocol));
```

Reply:
you mean ```const ports = get_default_ports(Number(port) || get_protocol_port(protocol));```
Agree it much cleaner - So as it already merged - will be part of my next PR